src/report/filter.rs

```diff
@@ -112,12 +114,31 @@ impl FilterRule {
                 Comparator::AtMost => report.confidence <= *level,
             },
             FilterRule::Key(key) => report.key == *key,
-            FilterRule::File(path) => {
-                report.pointers.iter().any(|pointer| pointer.loc.pathname().starts_with(path))
+            FilterRule::File(pattern) => {
+                report.pointers.iter().any(|pointer|
+                    pattern.matches_path(pointer.loc.pathname())
+                    || pointer.loc.pathname().starts_with(PathBuf::from(pattern.as_str())))
```

Comment:
Looks good :) There's just one issue here. The `PathBuf::from` call will create a new `PathBuf`, effectively copying the string, for every loop iteration.

You can instead use `Path::new` which is a cost free conversion that just wraps the existing string.